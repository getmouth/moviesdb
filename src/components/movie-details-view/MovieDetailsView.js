import React from 'react';
import { connect } from 'react-redux';
import { imageURL } from '../../actions/URL';
import './MovieDetailsView.scss';

type Props = {
  movie: Object,
}

const MovieDetailsView = (props: Props) => {
  const { movie } = props;
  return (
    <div className="MovieDetailsView">
      <h1>{movie.title}</h1>
      <img src={imageURL(movie.poster_path)} alt="" />
    </div>

  );
};

export default connect(({ movie }) => ({ movie }))(MovieDetailsView);
