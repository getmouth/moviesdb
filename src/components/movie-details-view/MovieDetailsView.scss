$background-color: #EFEFEF;

.MovieDetailsView {
  width: 100%;
  margin: auto;

  .DetailHeadFullWidth {
    width: 100%;
    background: lighten($background-color, 2%);
    padding: 80px 0;
    box-shadow: 0px 2px 5px 0px rgba(0, 0, 0, 0.3);
  }

  .DetailHead {
    display: flex;
    justify-content: center;
    width: 80%;
    margin: auto;
    color: #fff;

    .MovieInfo {
      width: 50%;
  
      h1, h3 {
        color: #fff;
      }
  
      h1 {
        span {
          margin-right: 10px;
        }
      }
  
      h3 {
        margin-bottom: 15px;
        line-height: 1.3;
        font-size: 21px;
      }
  
  
      .GenreLangCountryRow {
        .Genres {
          margin-bottom:  15px;
  
          span {
            background: #fff;
            color: rgba(0, 0, 0, 0.6);
            border-radius: 3px;
            font-style: italic;
            padding: 2px 5px;
          }
        }
  
        .Languages, .Country, .Genres {
          color: #fff;
          span {
            margin-right: 10px;
            font-size: 14px;
            box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2);
          }
        }
  
        .Languages, .Country {
          display: inline-block;
          span {
            padding: 4px 8px;
          }
  
        }
  
        .Country {
          margin-right: 35px;
        }
      }
    }
  
    .MovieImg {
      width: 300px;
      margin-right: 40px;
      img {
        width: 100%;
      }
    }
  
    .RatingRow {
      .RatingCircularBar {
        width: 60px;
        font-weight: bold;
      }
  
      p, .RatingCircularBar {
        display: inline-block;
        margin-right: 15px;
      }
    }
  }

  .CreditsRow {
    width: 80%;
    margin: auto;
  }


  // @media all and (max-width: 1200px) {
  
  //   .DetailHeadFullWidth {
  //     padding: 40px;
  //     height: auto;
  //   }

  //   .DetailHead {
  //     width: 100%;
  //   }

  //   .CreditsRow {
  //     padding: 0 40px;
  //     width: 100%;
  //   }

  // }

  @media all and (max-width: 900px) {
    .DetailHead {
      flex-direction: column;
      justify-content: center;
      width: 100%;

      
      .MovieImg {
        margin: auto;
        max-width: 90%;
      }

      .MovieInfo {
        width: 70%;
        margin: auto;
        text-align: center;
        

        
        .GenreLangCountryRow {
          text-align: center;
          .Languages, .Country {
            display: block;
            span {
              text-align: center;
            }
          }
        }
      }
    }
  }

  @media all and (max-width: 768px) {
    .CreditsRow {
      width: 100%;
    }
  }

  @media all and (max-width: 480px) {
    /* flips the MovieInfo and MovieImg
      on top of the other on mobile
    */
    .DetailHead {
      flex-direction: column;
      justify-content: center;
      width: 100%;
      .Languages, .Country {
        display: block;
      }

      .MovieImg {
        margin: auto;
        max-width: 90%;
      }

      .MovieInfo {
        width: 90%;
        margin: auto;

        .GenreLangCountryRow {
          .Languages, .Country {
            display: block;
            span {
              padding: 4px 4px;
              margin-bottom: 10px;
            }
          }
        }
        
      }
    }  
  }

}
