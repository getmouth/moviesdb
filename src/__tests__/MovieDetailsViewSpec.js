import React from 'react';
import renderer from 'react-test-renderer';
import { Route, BrowserRouter } from 'react-router-dom';
import { MovieDetailsView } from '../components/movie-details-view/MovieDetailsView';


describe('<MovieDetailsView />', () => {
  test('Should not Break When Movie id is passed', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Route path="movie/:5332-testmovie" component={MovieDetailsView} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
