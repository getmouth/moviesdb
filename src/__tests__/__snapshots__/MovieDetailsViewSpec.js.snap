// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetailsView /> Should not Break When Movie id is passed 1`] = `null`;
